packages/core/state/selection/selectors.ts

```diff
@@ -76,3 +77,37 @@ export const getUnavailableAnnotationsForHierarchy = createSelector(
             allAnnotations.filter((annotation) => !availableAnnotations.includes(annotation.name))
         )
 );
+
+// COMPOSED SELECTORS
+export const getSortedAnnotations = createSelector(
+    [getAnnotations, getRecentAnnotations],
+    (annotations: Annotation[], recentAnnotationNames: string[]) => {
+        // Create Array of annotations from recentAnnotationNames
+        const recentAnnotations = recentAnnotationNames.flatMap((name) =>
+            annotations.filter((annotation) => annotation.name === name)
+        );
+
+        // get the File name annotaion in a list (if Present)
+        const fileNameAnnotation = annotations.filter(
```

Comment:
I think you are looking for `find` here which returns the first element that matches the condition found or `undefined` if none are found

Reply:
This should make sure this isn't undefined